build/tools/SharedCodeGenerator/VersionComparer.cs

```diff
@@ -0,0 +1,52 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace SharedCodeGenerator
+{
+    internal class VersionComparer : IComparer<string>
```

Comment:
Ah man, I just stumbled across the https://github.com/microsoft/Oryx/blob/main/src/BuildScriptGenerator.Common/VersionInfo.cs class, which I can probably steal to do this in a cleaner way

Reply:
I wrote a `SemanticVersionComparer` in Terrapin that we could easily port over if you want to reuse that code.

Reply:
@cormacpayne As long as it can support some non-semantic versions, such as golang which we have as 1.17 (no patch version). I was able to modify the VersionInfo to fallback to the built in Version class for non-semantic-versioned platforms (right now just golang), I just haven't pushed any changes quite yet.